src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -705,25 +738,26 @@ class FastSync(
       !syncState.anythingQueued &&
       assignedHandlers.isEmpty
     }
-  }
 
-  private def updateBestBlockIfNeeded(receivedHashes: Seq[ByteString]): Unit = {
-    val fullBlocks = receivedHashes.flatMap { hash =>
-      for {
-        header <- blockchain.getBlockHeaderByHash(hash)
-        _ <- blockchain.getBlockBodyByHash(hash)
-        _ <- blockchain.getReceiptsByHash(hash)
-      } yield header
-    }
+    private def updateBestBlockIfNeeded(receivedHashes: Seq[ByteString]): Unit = {
+      val fullBlocks = receivedHashes.flatMap { hash =>
```

Comment:
It will probably come out during testing, but taking into account all network varaibility and many peers, is it possible that we will miss some hashes in cache during this update? 
It is just food for thought as i am still trying to figure out all corner cases myself :)

Reply:
I will rewrite this code to more defensive one